File: SortAlgorithms/bubble_sort.cpp
#include <iostream>
using namespace std;


void Swap(int *a,int *b){
	int temp = *a;
	*a = *b;
	*b = temp;
}

void BubbleSort(int array[],int size){
	bool swapped=false;
	for (int i = 0; i < size ; ++i){
		for (int j = 0; j < size-i; ++j){
			if (array[j] > array[j+1]){
				Swap(&array[j],&array[j+1]);
				swapped = true;
			}
		}
		if (swapped == false){
			break;
		}
	}
}


void Print(int array[],int size){
	for (int i = 0; i < size; ++i){
		cout<<array[i]<<" ";
	}
	cout<<endl;
}

int main(){
	int array[]={2,3,4,5,1};
	Print(array,5);
	BubbleSort(array,5);
	Print(array,5);
}

File: Web/BubbleSort/BubbleSort.js


swap(var x, var y) {
    var z = x;
    x = y;
    y = z;
}

function BubbleSort(array) {
    let size = array.lenght;
    for(let i = 0; i <= size; i++) {
        bool swaped = false;
        for(let j = 0; j <= size - i; i++) {
           if(array[j] > array[j+1]){
           swap(array[j], array[j+1]);
            swaped = true;
           }        
        }
         if (swapped == false) {
             return array;
         }
    }
}

var array = [87, 45, 98, 23, 15, 64, 96];
console.log(array);

BubbleSort(array);
console.log(array);